src/nvim/path.c

```diff
@@ -2186,9 +2186,16 @@ static int path_get_absolute_path(const char_u *fname, char_u *buf,
 
 /// Check if the given file is absolute.
 ///
-/// This just checks if the file name starts with '/' or '~'.
 /// @return `TRUE` if "fname" is absolute.
 int path_is_absolute_path(const char_u *fname)
 {
+#ifdef WIN32
+  // A name like "d:/foo" and "//server/share" is absolute
+  return ((isalpha(fname[0]) && fname[1] == ':'
```

Comment:
Vim's `mch_isFullName` doesn't check isalpha. Apparently old versions of Windows [supported non-alpha drive letters](http://stackoverflow.com/a/4652637/152142).